scheduler/scheduler.go

```diff
@@ -385,6 +406,32 @@ func (s *scheduler) startTask(id, source string) []serror.SnapError {
 		return errs
 	}
 
+	taskToPlugin := make(map[string][]core.Plugin)
+	for k := range clients {
+		cps := returnCorePlugin(clients[k])
+		taskToPlugin[k] = cps
+		mgr, err := t.RemoteManagers.Get(k)
+		// If we can't get any manager for this, no reason to continue
+		if err != nil {
+			return []serror.SnapError{serror.New(err)}
+		}
+		errs := mgr.SubscribeDeps(t.ID(), []core.Metric{}, cps)
```

Comment:
We do in this commit but I am working on an update that will make it unneeded and imo more intuitive.